src/editor/EditorManager.js

```diff
@@ -779,6 +778,25 @@ define(function (require, exports, module) {
         }
     }
 
+    function findProvider(editor,provider) {
+        var providers = (provider)?editProvider:docProvider,
+            i,
+            providerRet;
+
+        providerRet = providers(editor);
+        if (providerRet) {
+            return true;
+        }
+        return false;
+    }
+
+    function registerEditProvider(provider){
```

Comment:
This and the code below it seems wrong to me.  There is already code to register providers elsewhere.  Why aren't we just hooking into that, and allowing it to ask each provider if they have anything to show?